Bottom: dba80c04ea7d6148b0bec40a18bb1911a48fba34
Top:    03b18e8aab4a45da539dc081626c2faf86362a91
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 12:53:53 +0100

hid/gtk: Turn on defines to debug a potential GTK-3.0 build


---

diff --git a/configure.ac b/configure.ac
index 71ab1c4..d2e2eeb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -770,6 +770,9 @@ $GTKGLEXT_PKG_ERRORS])]
 	GTKGLEXT_VER=`$PKG_CONFIG gtkglext-1.0 --modversion`
 	fi
 
+	GTK_CFLAGS="$GTK_CFLAGS -DGSEAL_ENABLE"
+#	GTK_CFLAGS="$GTK_CFLAGS -DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED -DGSEAL_ENABLE"
+
 	;;
 
       gcode|nelma|png )
diff --git a/src/hid/gtk/ghid-main-menu.c b/src/hid/gtk/ghid-main-menu.c
index 6c0ca01..4ae8c9a 100644
--- a/src/hid/gtk/ghid-main-menu.c
+++ b/src/hid/gtk/ghid-main-menu.c
@@ -347,7 +347,7 @@ ghid_main_menu_real_add_resource (GHidMainMenu *menu, GtkMenuShell *shell,
               int pos;
 
               children = gtk_container_get_children (GTK_CONTAINER (shell));
-              pos = g_list_length (shell->children);
+              pos = g_list_length (children);
               g_list_free (children);
 
               if (strcmp (res->v[i].value, "@layerview") == 0)
